### DISCUSSION TITLE
How to customise the Portal web app

### DISCUSSION THREAD

**Discussion by FoxXxHater** [discord-tickets/bot#434] [Support (Q&A)] [answered]
Hello, 

i searched by my self, but not found anything. 
But does anyone know how to edit the settings website?

**Comment by rakosi2** [comment]
What settings are you looking for?

**Comment by FoxXxHater** [comment]
I dont know... everything. Like the colors and pictures, texts an everthying whats on the website. 
Translate is a good point too

**Comment by eartharoid** [comment]
The settings panel is not customisable as it is intended to be seen only by server administrators and infrequently, and adding translations would have made development take much longer. Translations may be added in the future though, but customizability is unlikely.

The user-facing portal part of the application will use your guild's logo and colours and will support translations when it is made.

**Reply by FoxXxHater** [reply]
Okay... 
So there is no way to edit the local web server?

**Reply by eartharoid** [reply]
It's not impossible but it's a lot of effort just to change a logo.

**Reply by FoxXxHater** [reply]
Okay. If it is not impossible I would like to try it. Which files would I need to edit?

**Reply by FoxXxHater** [reply]
I have already searched the files myself. However, the concept and the plaintexts not understood / found

**Reply by eartharoid** [reply, marked as answer]
You would need to clone https://github.com/discord-tickets/portal, `npm i`, make your changes to whatever you want, then `npm run build`.
Then use [`npm link`](https://docs.npmjs.com/cli/v9/commands/npm-link) followed by `npm i` in the bot directory to override the package.
If you're using a panel like Pterodactyl to host then you can't do that, so you would need to host the web app separately, as described [here](https://ptb.discord.com/channels/451745464480432129/1113787851377746021/1115651239062032495).

You can change the blurple colour in https://github.com/discord-tickets/portal/blob/main/tailwind.config.cjs#L8 but changing any other colours will take a lot of search and replace.

**Reply by eartharoid** [reply]
And if you do this, you won't be able to update the portal very easily when features are added.

**Reply by FoxXxHater** [reply]
Okay, thanks. That will be help

I run it on a linux ubuntu lxc container on proxmox

**Reply by FoxXxHater** [reply]
i have done my changes... and have 2 folders `/root/bot` and `root/portal`

I runed in the portal the `npm i` and 'npm run build'
After a looooong time i runed `npm link /root/portal` ind the bot directory... after that the 'npm i' in the bot folder

But my changes arent visable on the page. In the console yes but not on the settings page

Have i do it correct?

**Reply by eartharoid** [reply]
¯\_(ツ)_/¯

**Reply by FoxXxHater** [reply]
Okay, Okay... I understand XD

**Comment by FoxXxHater** [comment]
I am done... its working 🥇

**Reply by FoxXxHater** [reply]
Ore maybe not... now 😢

**Comment by FoxXxHater** [comment]
Okay, I'm done after all. Found my mistake. And even one in the original code.
I fully translated into german and custemized it to my own. 😄 

![image](https://github.com/discord-tickets/bot/assets/54017453/02557ec5-a214-4480-b4d0-0d55264cb9ec)

**Reply by eartharoid** [reply]
> And even one in the original code.

@FoxXxHater what was it?

**Reply by FoxXxHater** [reply]
Uhhhhhh
Wait a moment

**Reply by FoxXxHater** [reply]
In the file `src/routes/settings/[guild]/panels/+page.svelte`

![image](https://github.com/discord-tickets/bot/assets/54017453/4a625d80-36fe-46f7-96a9-c1892b2cdc8d)

in both helps are the same texts

**Comment by FoxXxHater** [comment]
@eartharoid 
I dont know if you recognized it... but i have a auto redirect from the index page to the `/settings` page maybe its an idea for the main code?

**Reply by eartharoid** [reply]
I could have but the homepage will have a purpose soon.

**Reply by FoxXxHater** [reply]
Okay, sounds good. I am excited

**Comment by WhoNix992** [comment]
@FoxXxHater muss ich irgendwas beachten ? ich bekomm es nicht nicht^^ hab das so gemacht wie beschrieben wurde bzw. rauszulesen ist aber er zeigt die standardseite an. Hab auch schon versucht dein bot und portal zu installieren aber da kommt beim bot direkt ein fehler hab dann versucht den standard bot und dein portal aber keine ahnung komm nicht auf den trichter wo mein fehler ist ^

**Reply by FoxXxHater** [reply]
Moin, ist bei mir schon eine weile her wann ich damit rumgespielt habe, aber vielleicht hilft dir mein Wiki? https://wiki.foxco-network.de/en/discord/dicord-ticket-bot-settings-portal

**Reply by FoxXxHater** [reply]
Zudem muss ich aber sagen, das ich meine Repos lange nicht aktualisiert habe, da ich jetzt andere Bots nutze

**Reply by WhoNix992** [reply]
> Moin, ist bei mir schon eine weile her wann ich damit rumgespielt habe, aber vielleicht hilft dir mein Wiki? https://wiki.foxco-network.de/en/discord/dicord-ticket-bot-settings-portal

Danke für deine Antwort. Hab schon alles getestet und probiert xD komme einfach nicht drauf wo mein Fehler ist. Klar hab auf deinem wiki geschaut und dementsprechend probiert aber klappt leider aufh nicht.

**Reply by FoxXxHater** [reply]
Dann bin ich leider auch erstmal raus. Wenn ich aus dem Urlaub zurück bin und ich Zeit finden sollte, kann ich meine Repos ja mal Updaten und gegebenen Falls das ganze Theater noch einmal durchgehen.

**Reply by WhoNix992** [reply]
ja easy genieß deinen urlaub =)

**Comment by ZkIsHere** [comment]
@FoxXxHater I followed your instructions on your site, but when I run the bot I get a bunch of errors like the one below, it's like the bot has every ability to give errors, this is just one of them
![Screenshot from 2025-05-27 22-47-00](https://github.com/user-attachments/assets/077418a3-4038-464b-b2b2-a8920e043dcd)

**Reply by FoxXxHater** [reply]
What kind of bot did you use?
Because mine are realy "old" and not updated to the lates official version

**Reply by ZkIsHere** [reply]
I use the latest version but the errors in the picture are errors when running the bot in your repo

**Comment by driggersj53-oss** [comment]
what file's Config what for the website, example, tailwind.config.js configures the theme
